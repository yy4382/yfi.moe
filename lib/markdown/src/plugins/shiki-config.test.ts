import rehypeStringify from "rehype-stringify";
import remarkParse from "remark-parse";
import remarkRehype from "remark-rehype";
import { unified } from "unified";
import { beforeEach, describe, expect, it, vi } from "vitest";

describe("shiki config", () => {
  let mod: typeof import("./shiki-config.js");

  beforeEach(async () => {
    vi.resetModules();
    mod = await import("./shiki-config.js");
  });

  const process = async (
    raw: string,
    shikiPreset: typeof mod.rehypeShikiPreset,
  ) => {
    const vfile = await unified()
      .use(remarkParse)
      .use(remarkRehype)
      .use(shikiPreset)
      .use(rehypeStringify)
      .process(raw);
    return String(vfile);
  };

  it("should work with 1 language", async () => {
    const raw = `
\`\`\`ts
console.log("Hello, world!");
\`\`\`
     `;
    const result = await process(raw, mod.rehypeShikiPreset);
    expect(result).toMatchSnapshot();
  });

  it("should work with 2 languages", async () => {
    const raw = `
\`\`\`ts
console.log("Hello, world!");
\`\`\`

\`\`\`python
print("Hello, world!");
\`\`\`
    `;
    const result = await process(raw, mod.rehypeShikiPreset);
    expect(result).toMatchSnapshot();
  });
  it("should work with 2 file with different languages", async () => {
    const raw1 = `
\`\`\`ts
console.log("Hello, world!");
\`\`\`
    `;
    const raw2 = `
\`\`\`python
print("Hello, world!");
\`\`\`
    `;
    const result1 = await process(raw1, mod.rehypeShikiPreset);
    expect(result1).toMatchSnapshot();
    const result2 = await process(raw2, mod.rehypeShikiPreset);
    expect(result2).toMatchSnapshot();
  });

  it("should not highlight inline code", async () => {
    const raw = `\`print("Hello, world!")\``;
    const result = await process(raw, mod.rehypeShikiPreset);
    expect(result).toMatchSnapshot();
  });
});
