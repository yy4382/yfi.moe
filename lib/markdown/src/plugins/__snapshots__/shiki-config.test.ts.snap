// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shiki config > should not highlight inline code 1`] = `"<p><code>print("Hello, world!")</code></p>"`;

exports[`shiki config > should work with 1 language 1`] = `"<pre class="shiki catppuccin-macchiato" style="background-color:#24273a;color:#cad3f5" tabindex="0"><code><span class="line"><span style="color:#CAD3F5">console</span><span style="color:#8BD5CA">.</span><span style="color:#8AADF4;font-style:italic">log</span><span style="color:#CAD3F5">(</span><span style="color:#A6DA95">"Hello, world!"</span><span style="color:#CAD3F5">)</span><span style="color:#939AB7">;</span></span></code></pre>"`;

exports[`shiki config > should work with 2 file with different languages 1`] = `"<pre class="shiki catppuccin-macchiato" style="background-color:#24273a;color:#cad3f5" tabindex="0"><code><span class="line"><span style="color:#CAD3F5">console</span><span style="color:#8BD5CA">.</span><span style="color:#8AADF4;font-style:italic">log</span><span style="color:#CAD3F5">(</span><span style="color:#A6DA95">"Hello, world!"</span><span style="color:#CAD3F5">)</span><span style="color:#939AB7">;</span></span></code></pre>"`;

exports[`shiki config > should work with 2 file with different languages 2`] = `"<pre class="shiki catppuccin-macchiato" style="background-color:#24273a;color:#cad3f5" tabindex="0"><code><span class="line"><span style="color:#F5A97F;font-style:italic">print</span><span style="color:#939AB7">(</span><span style="color:#A6DA95">"Hello, world!"</span><span style="color:#939AB7">)</span><span style="color:#CAD3F5">;</span></span></code></pre>"`;

exports[`shiki config > should work with 2 languages 1`] = `
"<pre class="shiki catppuccin-macchiato" style="background-color:#24273a;color:#cad3f5" tabindex="0"><code><span class="line"><span style="color:#CAD3F5">console</span><span style="color:#8BD5CA">.</span><span style="color:#8AADF4;font-style:italic">log</span><span style="color:#CAD3F5">(</span><span style="color:#A6DA95">"Hello, world!"</span><span style="color:#CAD3F5">)</span><span style="color:#939AB7">;</span></span></code></pre>
<pre class="shiki catppuccin-macchiato" style="background-color:#24273a;color:#cad3f5" tabindex="0"><code><span class="line"><span style="color:#F5A97F;font-style:italic">print</span><span style="color:#939AB7">(</span><span style="color:#A6DA95">"Hello, world!"</span><span style="color:#939AB7">)</span><span style="color:#CAD3F5">;</span></span></code></pre>"
`;
